### PR TITLE
Update changelog for 0.40.0rc1

### PR DESCRIPTION
**Pull request by sklam**

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,3 +1,54 @@
+v0.40.0rc1 (March 30, 2023)
+---------------------------
+
+This release predominantly upgrades to LLVM 14 and Python 3.11.
+Minimum supported python version is bumped to Python 3.8.
+
+Note: We discovered a bug in LLVM RuntimeDyldELF Aarch64 platform that 
+can cause segfaults when cross module symbols are linked. It is necessary
+for JIT users to build LLVM with the patch added in `PR#926 <https://github.com/numba/llvmlite/pull/926>`_.
```
Comment (stuartarchibald) [inline CHANGE_LOG:9]: line wrap to 80 chars?

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,3 +1,54 @@
...
+
+* `apmasell <https://github.com/apmasell>`_
+* `bslatkin <https://github.com/bslatkin>`_
+* `esc <https://github.com/esc>`_
+* `gmarkall <https://github.com/gmarkall>`_
+* `jakirkham <https://github.com/jakirkham>`_
+* `jamesobutler <https://github.com/jamesobutler>`_
+* `mbargull <https://github.com/mbargull>`_
+* `sklam <https://github.com/sklam>`_
+* `stuartarchibald <https://github.com/stuartarchibald>`_
```
Comment (stuartarchibald) [inline CHANGE_LOG:48]: This will need looking at once the list of included PRs is finalised.

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,3 +1,54 @@
+v0.40.0rc1 (March 30, 2023)
+---------------------------
+
+This release predominantly upgrades to LLVM 14 and Python 3.11.
+Minimum supported python version is bumped to Python 3.8.
```
Comment (stuartarchibald) [inline CHANGE_LOG:5]: There's also a lot more pass bindings added in this release (#827).

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch @sklam. There's a few minor things to look at, but the biggest change required is to fix the PR list to only include PRs that are new to this release. Thanks again!

**Comment by sklam** [comment]
TODO: 

- [ ] add https://github.com/numba/llvmlite/pull/931

**Comment by stuartarchibald** [comment]
> All suggestion fixed.

Thanks, I'll take a look. Does https://github.com/numba/llvmlite/pull/930#discussion_r1151839088 still need addressing?

> Sorry, i completely forgotten to check PRs duplication.

No problem, fixing the script to do this automatically is something to do as part of increasing release automation.

File: CHANGE_LOG [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+* PR `#903 <https://github.com/numba/llvmlite/pull/903>`_: Fix flake8 config and style for flake8 6 (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#905 <https://github.com/numba/llvmlite/pull/905>`_: Add YouCompleteMe configuration file and ignore vim swap files (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#906 <https://github.com/numba/llvmlite/pull/906>`_: Replace importlib-resources legacy API use (`sklam <https://github.com/sklam>`_)
+* PR `#910 <https://github.com/numba/llvmlite/pull/910>`_: Aarch64 split build for LLVM14 (`sklam <https://github.com/sklam>`_)
+* PR `#921 <https://github.com/numba/llvmlite/pull/921>`_: Setup AzureCI to use py311 and llvm14  (`sklam <https://github.com/sklam>`_)
+* PR `#922 <https://github.com/numba/llvmlite/pull/922>`_: Fix AzureCI not using llvm14 on windows (`sklam <https://github.com/sklam>`_)
+* PR `#926 <https://github.com/numba/llvmlite/pull/926>`_: llvmdev recipe: Add patch that clears GOTOffsetMap (`apmasell <https://github.com/apmasell>`_ `gmarkall <https://github.com/gmarkall>`_ `sklam <https://github.com/sklam>`_)
+
```
Comment (stuartarchibald) [inline CHANGE_LOG:32]: Missing PRs:
* Maybe #920 (think it was merged via #926 so doesn't need including)?
* #931

File: CHANGE_LOG [hunk 5]
```diff
@@ -1,3 +1,47 @@
...
+`PR#926 <https://github.com/numba/llvmlite/pull/926>`_.
+
+Pull-Requests:
+
+* PR `#827 <https://github.com/numba/llvmlite/pull/827>`_: Add more LLVM pass bindings (`apmasell <https://github.com/apmasell>`_)
+* PR `#830 <https://github.com/numba/llvmlite/pull/830>`_: Add LLVM 14 support (`apmasell <https://github.com/apmasell>`_)
+* PR `#860 <https://github.com/numba/llvmlite/pull/860>`_: the git tag for the RC needs an rc1 suffix (`esc <https://github.com/esc>`_)
+* PR `#869 <https://github.com/numba/llvmlite/pull/869>`_: bump max Python version to 3.11 (`esc <https://github.com/esc>`_ `sklam <https://github.com/sklam>`_)
+
+* PR `#876 <https://github.com/numba/llvmlite/pull/876>`_: Remove `llvmlite.llvmpy` after deprecation (`apmasell <https://github.com/apmasell>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:19]:
```suggestion
* PR `#876 <https://github.com/numba/llvmlite/pull/876>`_: Remove ``llvmlite.llvmpy`` after deprecation (`apmasell <https://github.com/apmasell>`_)
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the update, a couple more minor things to resolve. I've now reconciled the milestone for 0.40.0 with what is present in this patch.

File: CHANGE_LOG [hunk 6]
```diff
@@ -1,3 +1,47 @@
...
+* `apmasell <https://github.com/apmasell>`_
+* `bslatkin <https://github.com/bslatkin>`_
+* `esc <https://github.com/esc>`_
+* `gmarkall <https://github.com/gmarkall>`_
+* `jamesobutler <https://github.com/jamesobutler>`_
+* `mbargull <https://github.com/mbargull>`_
+* `sklam <https://github.com/sklam>`_
+
+
+
```
Comment (stuartarchibald) [inline CHANGE_LOG:44]: There should be one empty line here to be consistent with the rest of the document.

File: CHANGE_LOG (hunk 3 above)
Comment (stuartarchibald) [reply CHANGE_LOG]: This needs to be addressed.

File: CHANGE_LOG (hunk 6 above)
Comment (sklam) [reply CHANGE_LOG]: fixed in https://github.com/numba/llvmlite/pull/930/commits/e8362019e9a288e0265035f80dda4026214b24e7

File: CHANGE_LOG (hunk 5 above)
Comment (sklam) [reply CHANGE_LOG]: fixed in https://github.com/numba/llvmlite/pull/930/commits/e8362019e9a288e0265035f80dda4026214b24e7

File: CHANGE_LOG [hunk 7]
```diff
@@ -1,3 +1,48 @@
+v0.40.0rc1 (March 30, 2023)
```
Comment (sklam) [inline CHANGE_LOG:1]:
```suggestion
v0.40.0rc1 (April 5, 2023)
```

File: CHANGE_LOG [hunk 8]
```diff
@@ -1,3 +1,48 @@
...
+---------------------------
```
Comment (sklam) [inline CHANGE_LOG:2]:
```suggestion
v0.40.0rc1 (April 5, 2023)
--------------------------
```

File: buildscripts/manylinux_2_17/README.md [hunk 9]
```diff
@@ -7,7 +7,7 @@ Run the script below to start docker off building `llvmdev` base from the curren
 
 - 32-bit linux: `./buildscripts/manylinux_2_17/docker_run_x32.sh build_llvmdev.sh`
 - 64-bit linux: `./buildscripts/manylinux_2_17/docker_run_x64.sh build_llvmdev.sh`
-- aarch64 linux: `./buildscripts/manylinux_2_17/docker_run_aaarch64.sh build_llvmdev.sh`
+- aarch64 linux: `./buildscripts/manylinux_2_17/docker_run_aarch64.sh build_llvmdev.sh`
```
Comment (stuartarchibald) [inline buildscripts/manylinux_2_17/README.md:10]: Was it intended for this change to be present in this PR?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the update. A question to answer and a typo else looks good. I've checked the rendered build on RTD and it looks ok on inspection.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**